tests/check.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+#include "saiattributelist.h"
+
+struct Check {
+    static bool AttrListEq(sai_object_type_t objecttype, const std::vector<sai_attribute_t>& act_attr_list, /*const*/ SaiAttributeList& exp_attr_list)
+    {
+        if (act_attr_list.size() != exp_attr_list.get_attr_count()) {
+            return false;
+        }
+
+        for (uint32_t i = 0; i < exp_attr_list.get_attr_count(); ++i) {
+            sai_attr_id_t id = exp_attr_list.get_attr_list()[i].id;
+            auto meta = sai_metadata_get_attr_metadata(objecttype, id);
+
+            assert(meta != nullptr);
+
+            // The following id can not serialize, check id only
+            if (id == SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE) {
+                if (id != act_attr_list[i].id) {
+                    auto meta_act = sai_metadata_get_attr_metadata(objecttype, act_attr_list[i].id);
+
+                    if (meta_act) {
+                        std::cerr << "AttrListEq failed\n";
+                        std::cerr << "Actual:   " << meta_act->attridname << "\n";
+                        std::cerr << "Expected: " << meta->attridname << "\n";        
+                    }
+                }
+
+                continue;
+            }
+
+            char act_buf[0x4000];
```

Comment:
>act_buf [](start = 17, length = 7)

Suggest not to create large array in stack.

Reply:
Yes, shall not create large array in stack. I will fix it.